src/python/WMCore/MicroService/MSTransferor/DataStructs/DQMHarvestWorkflow.py

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+Workflow object representing how DQMHarvest workflows have to be dealt
+with in the input data placement (MSTransferor)
+"""
+
+from WMCore.MicroService.MSTransferor.DataStructs.Workflow import Workflow
+
+
+class DQMHarvestWorkflow(Workflow):
+    """
+    Class to represent a DQMHarvest workflow in the context
+    of input data placement in MSTransferor
+    """
+
+    def getInputData(self):
+        """
+        Returns all the primary data that has to be locked and
+        transferred with Rucio.
+        :return: a list of unique block names
+        """
+        blockList = list(self.getPrimaryBlocks())
+        totalBlockSize = sum([blockInfo['blockSize'] for blockInfo in self.getPrimaryBlocks().values()])
+        return blockList, totalBlockSize
+
+    def getRucioGrouping(self):
+        """
+        Returns the rucio rule grouping for DQMHarvest workflows,
+        which must hold all the input data under the same RSE in
+        order to harvest full stats.
+
+        :return: a string with the required DID grouping
+        """
+        return "ALL"
```

Comment:
wouldn't it be better to have this one configurable?

Reply:
It would for the MSTransferor. But at this object layer - and given that there are different objects with different attributes/values - I think having it as a constant is a good compromise without making this code more complex.

If I refactor the code and move it outside of the Workflow object, then it's much more doable. Let me give it further thought.